osbenchmark/aggregator.py

```diff
@@ -0,0 +1,216 @@
+from typing import Any, Dict, List, Union
+import uuid
+
+from osbenchmark.metrics import FileTestExecutionStore
+from osbenchmark import metrics, workload, config
+
+class Aggregator:
+    def __init__(self, cfg, test_executions_dict):
+        self.config = cfg
+        self.test_executions = test_executions_dict
+        self.accumulated_results: Dict[str, Dict[str, List[Any]]] = {}
+        self.accumulated_iterations: Dict[str, int] = {}
+
+    # count iterations for each operation in the workload
+    def iterations(self) -> None:
+        loaded_workload = workload.load_workload(self.config)
+        for task in loaded_workload.test_procedures:
+            for operation in task.schedule:
+                operation_name = operation.name
+                iterations = operation.iterations or 1
+                self.accumulated_iterations.setdefault(operation_name, 0)
+                self.accumulated_iterations[operation_name] += iterations
+
+    # accumulate metrics for each task from test execution results
+    def results(self, test_execution: Any) -> None:
```

Comment:
Similar to `iterations()`, I'd make this more descriptive to improve clarity.  Maybe substitute with `gather_results()`, `accumulate_results()`, or `add_to_accumulated_results()` ?

Reply:
After we verify that all test execution ids have the same test procedure, we'll need to add test_procedure to the config as well so that we can use that here to collect all the operations belonging to that test procedure.

Reply:
Fixed👍